test/ng/httpSpec.js

```diff
@@ -1380,8 +1380,49 @@ describe('$http', function() {
             expect(errCallback.calls.mostRecent().args[0]).toEqualMinErr('$http', 'baddata');
           });
 
-        });
+          it('should not throw an error if JSON is invalid but content-type is not application/json', function() {
+            var errCallback = jasmine.createSpy('error');
+            $httpBackend.expect('GET', '/url').respond('{abcd}', {'Content-Type': 'text/plain'});
+
+            $http.get('/url').then(callback).catch(errCallback);
+            $httpBackend.flush();
+
+            expect(callback).toHaveBeenCalledOnce();
+            expect(errCallback).not.toHaveBeenCalled();
+          });
 
+          it('should not throw an error if JSON is invalid but content-type is not specified', function() {
+            var errCallback = jasmine.createSpy('error');
+            $httpBackend.expect('GET', '/url').respond('{abcd}');
+
+            $http.get('/url').then(callback).catch(errCallback);
+            $httpBackend.flush();
+
+            expect(callback).toHaveBeenCalledOnce();
+            expect(errCallback).not.toHaveBeenCalled();
+          });
+
+          it('should return response unprocessed if JSON is invalid but content-type is not application/json', function() {
```

Comment:
The "return response unprocessed" tests could be merged with their "not throw an error"counterparts, but I am fine keeping them as is too.